### PR TITLE
Retry S3 PutObject request after REQUEST_TIMEOUT error.

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category:
- Not for changelog

Retry S3 PutObject request after REQUEST_TIMEOUT error

### CODE REVIEW

File: src/IO/S3/copyS3File.cpp [hunk 1]
```diff
@@ -524,6 +524,14 @@ namespace
                     continue; /// will retry
                 }
 
+                if ((outcome.GetError().GetErrorType() == Aws::S3::S3Errors::REQUEST_TIMEOUT) && (retries < max_retries))
```
Comment (vitlibar) [inline src/IO/S3/copyS3File.cpp:527]: Sometimes AWS S3 server respond on `PutObject` with the following error
```
AWSClient: AWSErrorMarshaller: Encountered AWSError 'RequestTimeout': Your socket connection to the server was not read from or written to within the timeout period. Idle connections will be closed.
```

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 6029d31528db38af2cf43c80967dae94264c45ff with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49670/6029d31528db38af2cf43c80967dae94264c45ff/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Is it generic enough to close #49680?

File: src/IO/S3/copyS3File.cpp (hunk 1 above)
Comment (antonio2368) [reply src/IO/S3/copyS3File.cpp:527]: I'm not sure how upload will behave if it notices that it was already uploaded but we will get a new error in worst case.

**Review by antonio2368 — APPROVED** [review]

**Comment by vitlibar** [comment]
This fix is wrong.

**Pull request closed without merging**